gobblin-api/src/main/java/gobblin/configuration/WorkUnitState.java

```diff
@@ -107,10 +109,10 @@ public void setWorkingState(WorkingState state) {
   /**
    * Get the actual high {@link Watermark} as a {@link JsonElement}.
    *
-   * @return a {@link JsonElement} representing the actual high {@link Watermark}.
+   * @return a JSON string representing the actual high {@link Watermark}.
    */
-  public JsonElement getActualHighWatermark() {
-    return GSON.toJsonTree(getProp(ConfigurationKeys.WORK_UNIT_STATE_ACTUAL_HIGH_WATER_MARK_KEY));
+  public String getActualHighWatermark() {
```

Comment:
Can you keep the return type as `JsonElement` and use a `JsonParser` like you did in the other classes.